packages/react-data-grid/src/Row.js

```diff
@@ -138,7 +138,20 @@ const Row = React.createClass({
       return this.props.isSelected;
     } else if (typeof this.props.row.get === 'function') {
       val = this.props.row.get(key);
-    } else {
+    } else if (key.indexOf('.') >= 0) {
+      val = (function getVal(valchain, subkey) {
+        const tokenIdx = subkey.indexOf('.');
+        if (tokenIdx >= 0) {
+          const token = key.slice(0, tokenIdx);
+          const remain = key.slice(tokenIdx + 1, key.length);
+          if (valchain[token]) {
+            return getVal(valchain[token], remain);
```

Comment:
hey @matrixlice -- thanks for the contribution. a couple of quick thoughts:-

1 - please can you extract `getVal()` to a separate utility function so we can include tests against it?

2 - rather than using indexes, slices, and recursively calling `getVal()`, could we instead do `subkey.split('.')` and use `Array.map()` to iterate over that?

here's a similar example:-

```
function getValueForKey (key, model) {
  let returnVal = model
  const keyArray = key.split('.')
  keyArray.map(key => {
    if (returnVal[key]) {
      returnVal = returnVal[key]
    }
  })
  return returnVal
}
```

this avoids the inline fn definition, is a single call, and will safely handle cases with and without subkeys.

what do you think? thanks!